### PR TITLE
[ex_unit] Omit excluded tests from test count in CLI summary

### PR DESCRIPTION
**Pull request by kieraneglin**
Discussed here: https://groups.google.com/g/elixir-lang-core/c/xh-hMsDi7c4

This PR omits the count of excluded tests from the overall test count in the CLI summary report

### Example

**Before:** `27 tests, 3 failures, 14 excluded`
**After:** `13 tests, 3 failures, 14 excluded`

### CODE REVIEW

**Comment by ericmj** [comment]
Is it confusing that the failures are a subset of the number of tests but the excluded or not?

If the goal is to quickly figure out the number of passing tests I think printing that directly would be more hopeful:

```
27 tests, 10 passing, 3 failures, 14 excluded
```

**Comment by josevalim** [comment]
I am not sure that tests that have been excluded actually matter on the overall results. It is good to know that something was excluded  but otherwise not relevant? So we could show that information between parens.

We could also show how many passed instead of how many tests ran (and you could always compute the sum if you need it for some reason).

**Comment by kieraneglin** [comment]
I'm open to whatever approach is preferred! Personally, I haven't seen much value in knowing the overall count of tests (regardless of how many were excluded) but every use case is different.

Just let me know what the desired output is and I'll update the PR (:

**Comment by josevalim** [comment]
Now that I think about it, it probably makes sense to not have "excluded" as part of tests because they have been *excluded* after all.

**Comment by kieraneglin** [comment]
I see what you mean and I do recognize that you're told when you're only testing a certain tag/line number, but for me it's still a small QOL improvement to indicate that not all tests were run in the final summary. 

My preference is still `13 tests, 3 failures, 14 excluded`. I think this contains all relevant data at a glance. I'm not a fan of `27 tests, 10 passing, 3 failures, 14 excluded` but I can't articulate why - maybe it's because indicating how many _pass_ could be considered redundant?

**Comment by josevalim** [comment]
I will go ahead and merge this now that v1.18 is out. We will have 6 months to give this a try and feel how it works in practice.

File: lib/ex_unit/lib/ex_unit/cli_formatter.ex [hunk 1]
```diff
@@ -19,7 +19,7 @@ defmodule ExUnit.CLIFormatter do
       width: get_terminal_width(),
       slowest: opts[:slowest],
       slowest_modules: opts[:slowest_modules],
-      test_counter: %{},
+      test_counter: %{test: 0},
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/cli_formatter.ex:22]: Do we need this change?

File: lib/ex_unit/lib/ex_unit/cli_formatter.ex [hunk 2]
```diff
@@ -37,7 +37,7 @@ defmodule ExUnit.CLIFormatter do
   def handle_cast({:suite_finished, times_us}, config) do
     test_type_counts = collect_test_type_counts(config)
 
-    if test_type_counts > 0 && config.excluded_counter == test_type_counts do
+    if test_type_counts == 0 && config.excluded_counter > 0 do
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/cli_formatter.ex:40]:
```suggestion
    if test_type_counts == 0 and config.excluded_counter > 0 do
```

**Comment by kieraneglin** [comment]
@josevalim refactored!

Part of this was needing to add a default to `test_counter` of `%{test: 0}`. Without this, runs that had all tests excluded would have the output of `0 failures, 2 excluded` instead of the previous output of `0 tests, 0 failures, 2 excluded`. With this change, the side effect is now that test runs that are 100% doctests indicate that there are `0 tests` now.

Do you have a preference for approach?

**Comment by josevalim** [comment]
So instead of adding tests at the beginning, we should check if the map is empty at the end, and then add tests with a value of zero. Otherwise if I only run doctests, it will say 234 doctests, 0 tests.

File: lib/ex_unit/test/ex_unit_test.exs [hunk 3]
```diff
@@ -886,7 +887,7 @@ defmodule ExUnitTest do
         end)
 
       assert output =~ max_failures_reached_msg()
-      assert output =~ "\n4 tests, 0 failures, 1 excluded, 2 invalid, 1 skipped\n"
+      assert output =~ "\n3 tests, 0 failures, 1 excluded, 2 invalid, 1 skipped\n"
```
Comment (josevalim) [inline lib/ex_unit/test/ex_unit_test.exs:890]: Sorry, I have one last nitpick. In this large list, we have excluded, invalid, and skipped, and it is hard to know which ones are counted as tests. I think we should either move "excluded" as the last one or perhaps show it in parens, such as "1 skipped (1 excluded)". WDYT?

Comment (kieraneglin) [reply lib/ex_unit/test/ex_unit_test.exs:890]: Don't be sorry! I appreciate the patience (:

I tried both locally and I think moving it to the last of the list reads the best. I'll push that up shortly and see what you think!

File: lib/ex_unit/lib/ex_unit/cli_formatter.ex [hunk 4]
```diff
@@ -349,6 +350,10 @@ defmodule ExUnit.CLIFormatter do
         config.skipped_counter > 0,
         &(&1 <> ", " <> skipped("#{config.skipped_counter} skipped", config))
       )
+      |> if_true(
+        config.excluded_counter > 0,
+        &(&1 <> ", #{config.excluded_counter} excluded")
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/cli_formatter.ex:355]: WDYT about this?

```suggestion
        &(&1 <> " (#{config.excluded_counter} excluded)")
```

Then it will be very clear it is not part of the overall sum?

Comment (kieraneglin) [reply lib/ex_unit/lib/ex_unit/cli_formatter.ex]: I dig it! Made that change 🤙

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:
